crates/derive/src/stages/batch_queue.rs

```diff
@@ -151,7 +153,14 @@ where
                 batch.check_batch(&self.cfg, &self.l1_blocks, parent, &mut self.fetcher).await;
             match validity {
                 BatchValidity::Future => {
-                    remaining.push(batch.clone());
+                    // Drop Future batches post-holocene.
+                    //
+                    // See: <https://specs.optimism.io/protocol/holocene/derivation.html#batch-queue>
+                    if !self.cfg.is_holocene_active(origin.timestamp) {
+                        remaining.push(batch.clone());
+                    } else {
+                        warn!(target: "batch-queue", "[HOLOCENE] Dropping future batch with parent: {}", parent.block_info.number);
```

Comment:
We should also flush the prev stage if we drop something here, yes?

Reply:
I don't think so.